RestaurantsNearMe/Data/YelpRequest/YelpResult.swift

```diff
@@ -0,0 +1,44 @@
+//
+//  YelpResult.swift
+//  RestaurantsNearMe
+//
+//  Created by Colin Evans on 2024-01-22.
+//
+
+import Foundation
+import CoreLocation
+
+struct YelpResult: Decodable {
+  let businesses: [Restaurant]
+  let total: Int
+  let region: CLLocation
+  
+  enum CodingKeys: String, CodingKey {
+    case businesses, total, region
+  }
+  
+  init(from decoder: Decoder) throws {
+    let container = try decoder.container(keyedBy: CodingKeys.self)
+    businesses = try container.decode([Restaurant].self, forKey: .businesses)
+    total = (try? container.decodeIfPresent(Int.self, forKey: .total)) ?? 0
+    let location = try container.decode(Location.self, forKey: .region)
+    region = CLLocation(latitude: location.latitude, longitude: location.longitude)
+  }
+}
+
+private struct Location: Decodable {
+  let longitude: Double
+  let latitude: Double
+  
+  enum CodingKeys: CodingKey {
+    case center
+  }
+  
+  init(from decoder: Decoder) throws {
+    let container = try decoder.container(keyedBy: CodingKeys.self)
+    let values = try container.decodeIfPresent(Location.self, forKey: .center)
```

Comment:
Are you sure this will do what you think it will? Should this be decoding `latitude` and `longitude`, rather than itself? 🤔

Reply:
I was trying to decode this:
```
{
  "region": {
     "center": {
        "longitude": Double
        "latitude": Double   
      }
  }
} 
```
in particular the nested JSON object. Ideally I wouldn't want to create another intermediate object just to extract the latitude and longitude. Although looking at this implementation it's not the easiest to read/.

```
struct A: Decodable {
  let center: B
}

struct B: Decodable {
  let longitude: Double
  let latitude": Double   
}
```

Reply:
Why not add the extra layer? It's not like these are your domain/UI models, riiiight? 😄

Reply:
That's true. The only real reason was to not add additional bloat, but that's subjective anyway